### PR TITLE
Update elasticsearch diamond collector to be v5.0.0 compatible

### PR DESCRIPTION
**Pull request by kpid**
Updated the ES collector to be compatible with the new stats interface.

### CODE REVIEW

File: src/diamond/collectors/elasticsearch/elasticsearch.py [hunk 1]
```diff
@@ -23,10 +23,18 @@
     import simplejson as json
 
 import diamond.collector
+import diamond.convertor
 
 RE_LOGSTASH_INDEX = re.compile('^(.*)-\d\d\d\d\.\d\d\.\d\d$')
 
 
+def _normalize_metric_value(metric_name, value):
```
Comment (venkey-ariv) [inline src/diamond/collectors/elasticsearch/elasticsearch.py]: Can we add a comment here and where this is called? Explicitly spelling out that we report values in KB/Bytes would kill any confusion

File: src/diamond/collectors/elasticsearch/elasticsearch.py [hunk 2]
```diff
@@ -215,7 +224,7 @@ def collect_instance_index_stats(self, host, port, metrics):
                                 index['primaries'])
 
     def collect_instance(self, alias, host, port):
-        result = self._get(host, port, '_nodes/_local/stats?all=true', 'nodes')
+        result = self._get(host, port, '_nodes/_local/stats', 'nodes')
```
Comment (venkey-ariv) [inline src/diamond/collectors/elasticsearch/elasticsearch.py]: What is the impact of removing "?all=true"? Would this render this collector backwards incompatible?

File: src/diamond/collectors/elasticsearch/elasticsearch.py [hunk 3]
```diff
@@ -293,16 +302,29 @@ def collect_instance(self, alias, host, port):
...
-                             ['disk_reads'])
-            self._add_metric(metrics, 'disk.reads.size', fs_data,
-                             ['disk_read_size_in_bytes'])
-            self._add_metric(metrics, 'disk.writes.count', fs_data,
-                             ['disk_writes'])
-            self._add_metric(metrics, 'disk.writes.size', fs_data,
-                             ['disk_write_size_in_bytes'])
+        if 'fs' in data:
+            # elasticsearch >= 5.0.0
+            if 'io_stats' in data['fs'] and 'devices' in data['fs']['io_stats']:
```
Comment (venkey-ariv) [inline src/diamond/collectors/elasticsearch/elasticsearch.py]: and data['fs']['io_stats']['devices']

File: src/diamond/collectors/elasticsearch/elasticsearch.py [hunk 4]
```diff
@@ -293,16 +302,29 @@ def collect_instance(self, alias, host, port):
...
+                self._add_metric(metrics, 'disk.reads.count', fs_data,
+                                 ['read_operations'])
+                self._add_metric(metrics, 'disk.reads.size', fs_data,
+                                 ['read_kilobytes'])
+                self._add_metric(metrics, 'disk.writes.count', fs_data,
+                                 ['write_operations'])
+                self._add_metric(metrics, 'disk.writes.size', fs_data,
+                                 ['write_kilobytes'])
+            # elasticsearch < 5.0.0
+            elif 'data' in data['fs']:
```
Comment (venkey-ariv) [inline src/diamond/collectors/elasticsearch/elasticsearch.py]: and data['fs']['data']

File: src/diamond/collectors/elasticsearch/elasticsearch.py (hunk 3 above)
Comment (kpid) [inline src/diamond/collectors/elasticsearch/elasticsearch.py]: backward-compatible

File: src/diamond/collectors/elasticsearch/elasticsearch.py (hunk 4 above)
Comment (kpid) [inline src/diamond/collectors/elasticsearch/elasticsearch.py]: backward-compatible

**Comment by dichiarafrancesco** [comment]
It looks good to me. please as @venkey-ariv  suggested make sure you document the change so people won't get confused.

File: src/diamond/collectors/elasticsearch/elasticsearch.py (hunk 1 above)
Comment (kpid) [inline src/diamond/collectors/elasticsearch/elasticsearch.py]: This is done

**Pull request merged**